### PR TITLE
fix(task): subcommand parser skips global args

### LINKED ISSUE #15290: `-q` option position is inconsistent for the `task` subcommand
There is a tiny consistency bug in the CLI's `-q` option position for the `task` subcommand specifically. The Deno CLI itself (`deno [OPTIONS] [SUBCOMMAND]`) supports `-q` for most (all?) subcommands except `task`.

Consider the following:

```
$ echo '{"tasks": {"echo": "echo"}}' > deno.json
$ deno -q task echo
Task not found: task
Available tasks:
- echo
    echo
```

Instead, the `-q` option must appear _after_ the subcommand:

```
$ deno task -q echo

```

This is inconsistent with most other subcommands and the the CLI help documentation. Eg, `fmt` works as expected:

```
$ deno -q fmt
```

The expected supported format is always `deno [OPTIONS] [SUBCOMMAND]`. `deno -q task`, in this case.

### PR DESCRIPTION
**Pull request by Cre3per**
Fixes #15290

I decided to slice `args` in `flags_from_vec()`, rather than in `task_parse()`, to avoid passing `task_parse()` more data than is necessary. This makes it easier to see that `task_parse()` doesn't care about the "--quiet" in `deno --quiet task echo`.

Not happy with the way I'm slicing the subcommand's args from `args`. I couldn't find anything in clap to get the index of the subcommand name, or a function like `index_of()` for subcommands. Suggestions are welcome.

### CODE REVIEW

**Comment by niedzielski** [comment]
Hi @Cre3per! I'm not a maintainer just someone interested in seeing this patch merged! I just noticed that it looks like you have to sign the CLA if you want this code to be reviewed. Thanks either way!

**Comment by Cre3per** [comment]
Thanks for the reminder @niedzielski !

I pushed commits with a wrong email address, which I later amended, and thought the CLA checker was just slow to pick up the changes. I now force-pushed onto my fork.

**Comment by Cre3per** [comment]
@kt3k Github actions failure looks like #15385. Could you try re-running without changes?

File: cli/args/flags.rs [hunk 1]
```diff
@@ -2582,8 +2593,6 @@ fn task_parse(
   }
 
   if let Some(mut index) = matches.index_of("task_name_and_args") {
```
Comment (dsherret) [inline cli/args/flags.rs:2595]: This index being out of sync with the raw_args index seems a bit error prone. In the interest of merging this for the release tomorrow, I'm going to push a change that keeps these in sync.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks! Edit: and I reverted my commit.

File: cli/args/flags.rs (hunk 1 above)
Comment (dsherret) [reply cli/args/flags.rs:2584]: Ok, I couldn't figure it out in a reasonable time. I moved this current code to be localized and it seems to work in a bunch of scenarios I tried so let's land it. Thanks!

**Pull request merged**